app/controllers/project.py

```diff
@@ -283,6 +284,26 @@ def delete(self, project_id):
             ), 200
 
         except client.rest.ApiException as e:
+            if e.status == 404:
+                """
+                deletes apps and project from db if not found on the cluster
+                """
+                apps_list = project.apps
+                if apps_list:
+                    for app in apps_list:
+                        # delete the app from the database
+                        deleted = app.delete()
+
+                deleted = project.delete()
+
+                if not deleted:
+                    return dict(status='fail', message='deletion failed'), 500
```

Comment:
small nit here, would a status code of 500 be ideal for this. if so, any pointers as to why assuming its not a server error since we are handling the exception already

Reply:
At this point for the case of deleting a database record. If this doesn't happen it usually means that the backend server is unable to communicate with the `DB`, meaning its a server issue, that's why we use `500`

As for the `exception` we are handling the exception from the `Kubernetes client` .
And function `.delete()` will not raise an exception too because it handles the database exceptions with in it